clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -391,18 +392,21 @@ public KafkaProducer(Properties properties, Serializer<K> keySerializer, Seriali
             int retries = configureRetries(config, transactionManager != null, log);
             int maxInflightRequests = configureInflightRequests(config, transactionManager != null);
             short acks = configureAcks(config, transactionManager != null, log);
+            int deliveryTimeoutMs = configureDeliveryTimeout(config, log);
 
             this.apiVersions = new ApiVersions();
             this.accumulator = new RecordAccumulator(logContext,
                     config.getInt(ProducerConfig.BATCH_SIZE_CONFIG),
-                    this.totalMemorySize,
                     this.compressionType,
-                    config.getLong(ProducerConfig.LINGER_MS_CONFIG),
+                    config.getInt(ProducerConfig.LINGER_MS_CONFIG),
```

Comment:
Why changing linger to int ?

Reply:
we have `request.timeout.ms` and `delivery.timeout.ms` of `int` type. this is to make the type of `linger.ms` be consistent with other timeout related settings.